### PR TITLE
fix image inventory unit test. Buffer for ID was too short.

### PR DESCRIPTION
**Pull request by HenryRawas**
Fix for docker-provider unit test.

### CODE REVIEW

File: test/code/providers/Container_ImageInventory_Class_Provider_UnitTest.cpp [hunk 1]
```diff
@@ -155,14 +155,15 @@ class ImageInventoryTest : public CppUnit::TestFixture
 		// Should have same number of images
 		CPPUNIT_ASSERT_EQUAL(images.size(), context.Size());
 
-		wchar_t currentId[66];
+		wchar_t currentId[72];
```
Comment (jeffaco) [inline test/code/providers/Container_ImageInventory_Class_Provider_UnitTest.cpp]: I hate buffers like this (just big enough). How about something PLENTY big enough, like 128 or something?

File: test/code/providers/Container_ImageInventory_Class_Provider_UnitTest.cpp [hunk 2]
```diff
@@ -155,14 +155,15 @@ class ImageInventoryTest : public CppUnit::TestFixture
...
 		int imageCount = 0;
 
 		// Verify every field of every object
 		for (unsigned i = 0; i < images.size(); i++)
 		{
 			bool flag = false;
-			mbstowcs(currentId, cJSON_GetObjectItem(images[i], "InstanceID")->valuestring, 65);
+			int rc = mbstowcs(currentId, cJSON_GetObjectItem(images[i], "InstanceID")->valuestring, 71);
+			if (rc > 0 && rc < 72) currentId[rc] = 0;
```
Comment (jeffaco) [inline test/code/providers/Container_ImageInventory_Class_Provider_UnitTest.cpp]: Blech, 71 is a magic number. Please change to something like:

```
			int rc = mbstowcs(currentId, cJSON_GetObjectItem(images[i], "InstanceID")->valuestring, sizeof(currentID) - 1);
			if (rc > 0 && rc < sizeof(currentID)) currentId[rc] = 0;
```

This allows for easier changes to the buffer size down the road, should it ever be needed again.

**Comment by jeffaco** [comment]
Apply my changes above and update your PR. Then I'll take it and commit.

Or another option is to give me a firm reason why my suggestions are crazy and nuts! :smile:

File: test/code/providers/Container_ImageInventory_Class_Provider_UnitTest.cpp [hunk 3]
```diff
@@ -155,14 +155,15 @@ class ImageInventoryTest : public CppUnit::TestFixture
...
-		wchar_t currentId[66];
+		wchar_t currentId[128];
 		int imageCount = 0;
 
 		// Verify every field of every object
 		for (unsigned i = 0; i < images.size(); i++)
 		{
 			bool flag = false;
-			mbstowcs(currentId, cJSON_GetObjectItem(images[i], "InstanceID")->valuestring, 65);
+			int rc = mbstowcs(currentId, cJSON_GetObjectItem(images[i], "InstanceID")->valuestring, (sizeof(currentId)/sizeof(currentId[0])) - 1;
```
Comment (jeffaco) [inline test/code/providers/Container_ImageInventory_Class_Provider_UnitTest.cpp]: I think you have unbalanced parenthesis here. Please verify that, with your fix, it still compiles and it fixes the problem (i.e. the unit test that this affects is fixed).

**Comment by jeffaco** [comment]
One more iteration. Compile and test your changes please.

**Comment by jeffaco** [comment]
Signed off.

**Comment by jeffaco** [comment]
Is this ready to be merged? Please let me know, thanks!

**Comment by HenryRawas** [comment]
Yes, ready to be merged.

**Comment by jeffaco** [comment]
Committed to the master branch:

```
commit 74818dd (HEAD, origin/master, origin/HEAD, master)
Author: Jeff Coffler <jeff@taltos.com>
Date:   Wed Jun 1 14:54:45 2016

    Fix image inventory unit test: Buffer for ID was too short.
    
    These changes came from contractor; I'm just doing the merge to master.
```

**Pull request closed without merging**